src/globus_sdk/services/search/data.py

```diff
@@ -55,15 +55,16 @@ def set_advanced(self, advanced: bool) -> "SearchQuery":
         self["advanced"] = advanced
         return self
 
+    # TODO: type signature
     def add_facet(
         self,
         name,
         field_name,
-        type="terms",
+        type: str = "terms",
         size=None,
         date_interval=None,
         histogram_range=None,
-        **kwargs
+        **kwargs: Any
```

Comment:
This should probably become an `additional_fields` dict...

Why don't we kind of ignore `SearchData` for the moment, and I can circle back with a PR to tidy it up and add more annotations after this PR merges? That way, we don't get bogged down in oddities like this. Plus, `histogram_range` being a tuple is odd and makes me want to take another look (but maybe it's fine...).

Reply:
I just realized, rereading your original comments, that this is why we're failing the mypy untyped defs check.
I can queue up some suggested annotations like the above right now, if you like.

Reply:
If you have the time to add the annotations for the search modules, that would be stellar and, as you noticed, clear up the current warnings. Otherwise we can postpone the strict checks on those modules.

Reply:
I'll open a PR against your fork with the addition of these types. Hopefully that will get things passing.